### DISCUSSION TITLE
Prisma context in middlewares

### DISCUSSION THREAD

**Discussion by leo-petrucci** [prisma/prisma#6281] [Q&A] [answered]
I've been reading the [middleware docs page](https://www.prisma.io/docs/concepts/components/prisma-client/middleware) and thought this would be a good way to implement some functionality.

Essentially, when my API receives a request I want to:

- Check if the request contains Auth cookies
- Ensure the access token is not expired
- If it is expired, return a custom header informing the client

Usually I would be able to get my cookies like this:

```
      const Authorization = cookie.parse(context.req.headers['cookie'])
      const { accessToken } = Authorization
```

But I can't figure out how to access the `context` from the middleware. 

Any idea how I can achieve this?

**Comment by leo-petrucci** [comment, marked as answer]
So, I ended up using `graphql-middleware` and added a function to my middlewares there to do the same check. Works well.

**Reply by ryands17** [reply]
This will be the ideal way to do this as of now.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
